src/es_archiver/lib/indices/kibana_index.js

```diff
@@ -103,7 +104,7 @@ export async function migrateKibanaIndex({ client, log, kibanaPluginIds }) {
     },
     version: kibanaVersion,
     savedObjectSchemas: new SavedObjectsSchema(uiExports.savedObjectSchemas),
-    savedObjectMappings: uiExports.savedObjectMappings,
+    savedObjectMappings: convertLegacyMappings(uiExports.savedObjectMappings),
     savedObjectMigrations: uiExports.savedObjectMigrations,
     savedObjectValidations: uiExports.savedObjectValidations,
     callCluster: (path, ...args) => _.get(client, path).call(client, ...args),
```

Comment:
This is a missed blocker regarding saved object migration to new platform: ATM the `es-archiver`  has no way to be informed about mappings defined from NP (and it will be the same when we migrate schemas, migrations and validations).However I don't want to handle it in this PR (and I don't have a solution ATM) Created https://github.com/elastic/kibana/issues/55860 to discuss the issue.